src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -218,6 +218,12 @@ public async Task InitializeAsync(IEnumerable<FunctionMetadata> functions, Cance
             }
 
             _workerRuntime = _workerRuntime ?? _environment.GetEnvironmentVariable(EnvironmentSettingNames.FunctionWorkerRuntime);
+
+            if (functions != null)
+            {
+                _workerRuntime = _workerRuntime ?? Utility.GetWorkerRuntime(functions);
+            }
```

Comment:
There is a method inside GetWorkerRuntime() which throws exception if "functions" is null. That's why added an if check for not null.

Reply:
Update `GetWorkerRuntime` helper to take environment variable as precedence  -this will avoid line 220

Reply:
also, move null check on functions into `GetWorkerRuntime`

Reply:
updated the code here.